core/dbt/context/context_config.py

```diff
@@ -222,69 +189,107 @@ def translate_hook_names(self, project_dict):
             project_dict["post-hook"] = project_dict.pop("post_hook")
         return project_dict
 
-    def calculate_node_config_dict(
+    # RenderedConfigGenerator. Validation is performed, and a config object is returned.
+    def generate_node_config(
         self,
         config_call_dict: Dict[str, Any],
         fqn: List[str],
         resource_type: NodeType,
         project_name: str,
-        base: bool,
         patch_config_dict: Optional[dict] = None,
     ) -> Dict[str, Any]:
-        config = self.calculate_node_config(
+
+        config_cls = get_config_for(resource_type)
+        config_dict = self.combine_config_dicts(
             config_call_dict=config_call_dict,
             fqn=fqn,
             resource_type=resource_type,
             project_name=project_name,
-            base=base,
             patch_config_dict=patch_config_dict,
         )
+        fix_hooks(config_dict)
         try:
-            finalized = config.finalize_and_validate()
-            return finalized.to_dict(omit_none=True)
+            config_cls.validate(config_dict)
+            config_obj = config_cls.from_dict(config_dict)
```

Comment:
I'm noticing we are [no longer omitting `None` values](https://github.com/dbt-labs/dbt-core/pull/11051/files#diff-5b85f0049476b4454944daa5f50db50088ff1eb3b095d59ce6f1e393a088bc06L76) during construction here. 

Is the implication of that that even `None` configurations will make their way onto the parsed node representation + get serialized out? I don't think this is a huge deal, just trying to wrap my head around any potential downsides of the change.

Reply:
The config_dict is the merged dicts from all the various sources, and the method actually returns a config obj. It doesn't really matter whether you've omitted the None fields when constructing an object. This is called in build_config_dict, which converts it to a dictionary with omit_none, so it should actually be the same dictionary that was being returned before.